rotkehlchen/db/drivers/server.py

```diff
@@ -0,0 +1,227 @@
+        connect_func = sqlite3.connect if call_data.connection_type.value == DBConnectionType.GLOBAL.value else sqlcipher.connect  # noqa: E501  # pylint: disable=no-member
+        self.db_connections[call_data.db_path] = connect_func(*call_data.args, **call_data.kwargs)
+
+    def _handle_query(
+            self,
+            db_connection: UnderlyingConnection,
+            call_data: ZMQCallData,
+    ) -> ZMQReturnData:
+        """Handle a query by getting the starting reference (cursor or connection)
+        and calling the method on it.
+
+        May Raise:
+            KeyError: If the cursor name is specified but no cursor is found.
+            AttributeError: If the method is not found on the starting reference."""
+        # Get the starting reference (cursor or connection).
+        starting_reference: UnderlyingConnection | UnderlyingCursor
+        if call_data.cursor_name is not None:
+            # If a cursor name is specified, get the cursor from db_cursors.
+            starting_reference = self.db_cursors[call_data.db_path][call_data.cursor_name]
+            if starting_reference is None:
+                # If no cursor is found, raise a KeyError.
+                raise KeyError(f'No cursor for {call_data.db_path} found')
+        else:
+            # If no cursor name is specified, use the database connection.
+            starting_reference = db_connection
+
+        # Call the method on the starting reference with the specified arguments
+        return getattr(starting_reference, call_data.method.serialize())(*call_data.args, **call_data.kwargs)  # noqa: E501
+
+    def listen(self) -> None:
+        while True:
+            try:
+                call_data: ZMQCallData = self.zmq_connection.recv_pyobj()
+                if call_data.method.value == DbMethod.INITIALIZE.value:
+                    self._handle_connection(call_data)
+                    self.zmq_connection.send_pyobj(None)
+                    continue
+
+                try:
+                    db_connection = self.db_connections[call_data.db_path]
+                except KeyError:
+                    self.zmq_connection.send_pyobj(ZMQReturnData(
+                        result=None,
+                        error=KeyError(f'No DB connection for {call_data.connection_type} found'),
+                    ).serialize())
+                    continue
+
+                if call_data.method.value == DbMethod.OPEN_CURSOR.value:
+                    self.db_cursors[call_data.db_path][call_data.cursor_name or 'default'] = db_connection.cursor()  # noqa: E501
+                    self.zmq_connection.send_pyobj(None)
+                    continue
+                if call_data.method.value == DbMethod.CLOSE_CURSOR.value:
+                    del self.db_cursors[call_data.db_path][call_data.cursor_name or 'default']
+                    self.zmq_connection.send_pyobj(None)
+                    continue
```

Comment:
in that case, it will raise a KeyError because it will try to get the cursor of that db_path and find nothing. This error will be caught below and returned back to the client, and then raised there.